client/src/main/kotlin/tech/figure/classification/asset/client/client/impl/DefaultACExecutor.kt

```diff
@@ -168,7 +170,17 @@ class DefaultACExecutor(
             mode = options.broadcastMode,
         ).also { response ->
             if (response.txResponse.code != 0) {
-                throw IllegalStateException("Asset classification contract execution failed with message:${System.lineSeparator()}${response.txResponse.rawLog}")
+                throw IllegalStateException(
+                    buildLogMessage(
+                        "Asset classification contract execution failed",
+                        "raw log" to response.txResponse.rawLog,
+                        "hash" to response.txResponse.txhash,
+                        "status code" to response.txResponse.code,
+                        "height" to response.txResponse.height,
+                        "message" to Base64.getDecoder().decode(msg.msg.toByteArray()),
```

Comment:
Is this the best way to stringify `MsgExecuteContract`?

Reply:
To add on, this might end up being an enormous amount of data in log messages.  Just having the hash/height/signer in here would probably be sufficient, and then you could just follow up any errors by pulling up the tx on explorer.  This is fine as it is, but if the logs get enormous, drawing it back to just those elements would be a good compromise IMO

Reply:
sure thing, removed the message